pkg/diagnostics/utils/path_normalization_utils.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import "net/http"
+
+type PathNormalizationRW struct {
+	http.ResponseWriter
+	NormalizedPath *string
```

Comment:
why we need this to be a pointer? I did not see a check against nil or something like it that would warrant it, but maybe I missed it. Asking as you already use pointers against the struct.

Reply:
I used the pointer to pass by reference instead of passing by value and with that, we can perform the path transformation by modifying the var.

Reply:
thanks for the tip! as you pointed out the issue was on the return value. Fixed in this commit: https://github.com/dapr/dapr/pull/7746/commits/8dc24c680ddb1b98d850a101cd258802a06101b4